app/src/main/java/org/mozilla/fenix/collections/CollectionCreationController.kt

```diff
@@ -0,0 +1,202 @@
+
+@file:Suppress("TooManyFunctions")
+
+package org.mozilla.fenix.collections
+
+import androidx.annotation.VisibleForTesting
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import mozilla.components.browser.session.SessionManager
+import mozilla.components.feature.tab.collections.TabCollection
+import mozilla.components.feature.tabs.TabsUseCases
+import org.mozilla.fenix.R
+import org.mozilla.fenix.components.Analytics
+import org.mozilla.fenix.components.TabCollectionStorage
+import org.mozilla.fenix.components.metrics.Event
+import org.mozilla.fenix.home.sessioncontrol.Tab
+import org.mozilla.fenix.home.sessioncontrol.toSessionBundle
+
+interface CollectionCreationController {
+
+    fun saveCollectionName(tabs: List<Tab>, name: String)
+
+    fun renameCollection(collection: TabCollection, name: String)
+
+    /**
+     * See [CollectionCreationInteractor.onBackPressed]
+     */
+    fun backPressed(fromStep: SaveCollectionStep)
+
+    /**
+     * See [CollectionCreationInteractor.selectAllTapped]
+     */
+    fun selectAllTapped()
+
+    /**
+     * See [CollectionCreationInteractor.deselectAllTapped]
+     */
+    fun deselectAllTapped()
+
+    /**
+     * See [CollectionCreationInteractor.close]
+     */
+    fun close()
+
+    fun selectCollection(collection: TabCollection, tabs: List<Tab>)
+
+    /**
+     * See [CollectionCreationInteractor.saveTabsToCollection]
+     */
+    fun saveTabsToCollection(tabs: List<Tab>)
+
+    fun addNewCollection()
+
+    fun addTabToSelection(tab: Tab)
+
+    fun removeTabFromSelection(tab: Tab)
+}
+
+class DefaultCollectionCreationController(
+    private val store: CollectionCreationStore,
+    private val dismiss: () -> Unit,
+    private val analytics: Analytics,
+    private val tabCollectionStorage: TabCollectionStorage,
+    private val tabsUseCases: TabsUseCases,
+    private val sessionManager: SessionManager,
+    private val lifecycleScope: CoroutineScope
+) : CollectionCreationController {
+    override fun saveCollectionName(tabs: List<Tab>, name: String) {
+        dismiss()
+
+        val sessionBundle = tabs.toList().toSessionBundle(sessionManager)
+        lifecycleScope.launch(Dispatchers.IO) {
+            tabCollectionStorage.createCollection(name, sessionBundle)
+        }
+
+        analytics.metrics.track(
+            Event.CollectionSaved(normalSessionSize(sessionManager), sessionBundle.size)
+        )
+
+        closeTabsIfNecessary(tabs, sessionManager, tabsUseCases)
+    }
+
+    override fun renameCollection(collection: TabCollection, name: String) {
+        dismiss()
+        lifecycleScope.launch(Dispatchers.IO) {
+            tabCollectionStorage.renameCollection(collection, name)
+            analytics.metrics.track(Event.CollectionRenamed)
+        }
+    }
+
+    override fun backPressed(fromStep: SaveCollectionStep) {
+        handleBackPress(fromStep)
+    }
+
+    override fun selectAllTapped() {
```

Comment:
nit:In controllers, these should be named what the methods are doing, not what is triggering it.